core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
+                    'rule.')
+        # The Jinja representation of the answer is:
+        #   {{answer.proof_string}}
+
+        # Because the rule implies the proof was correct, half of the
+        # CheckedProof structure does not need to be saved. The remaining
+        # structure consists of three strings: assumptions_string,
+        # target_string, and proof_string. The latter is already available
+        # as the answer_str.
+        if answer_str == None:
+            return (
+                None,
+                'Failed to recover CheckedProof answer: \'%s\'' % answer_str)
+
+        # assumptions_string and target_string come from the assumptions and
+        # results customized to this particular LogicProof instance.
+        question_details = (
+            interaction.customization_args['question']['value'])
+        assumptions = question_details['assumptions']
+        results = question_details['results']
+
+        expressions = []
+        top_types = []
+        for assumption in assumptions:
+            expressions.append(assumption)
+            top_types.append('boolean')
+        expressions.append(results[0])
+        top_types.append('boolean')
+        operators = AnswerMigrationJob._BASE_STUDENT_LANGUAGE['operators']
+
+        if len(assumptions) <= 1:
+            assumptions_string = (
+                AnswerMigrationJob._display_expression_array(
+                    assumptions, operators))
+        else:
+            assumptions_string = '%s and %s' % (
+                AnswerMigrationJob._display_expression_array(
+                    assumptions[0:-1], operators),
+                AnswerMigrationJob._display_expression_helper(
+                    assumptions[-1], operators, 0))
+
+        target_string = AnswerMigrationJob._display_expression_helper(
+            results[0], operators, 0)
+
+        # The resulting answer is only 'Correct' if it is not a default answer,
+        # since if it were correct then it would have been matched to the
+        # 'Correct' rule type.
+        correct = rule_str != cls._DEFAULT_RULESPEC_STR
+        if correct:
+            return cls._normalize_raw_answer_object(objects.CheckedProof, {
+                'assumptions_string': assumptions_string,
+                'target_string': target_string,
+                'proof_string': answer_str,
+                'correct': True
+            }, answer_str)
+        else:
+            return cls._normalize_raw_answer_object(objects.CheckedProof, {
+                'assumptions_string': assumptions_string,
+                'target_string': target_string,
+                'proof_string': answer_str,
+                'correct': False,
+                'error_category': '',
+                'error_code': '',
+                'error_message': '',
+                'error_line_number': -1
+            }, answer_str)
+
+    @classmethod
+    def _cb_reconstitute_math_expression_input(
```

Comment:
Well we also want to migrate multiple choice, too #857. I have an issue for designing a general purpose answer migration system in #1733. It's a trickier problem than the exploration migration due to contention and lack of versioned entities, but I don't want to solve it as part of this one-off migration because it's going to lead to a much more convoluted system.